api/transport/router.go

```diff
@@ -59,8 +60,11 @@ func (p Procedure) MarshalLogObject(enc zapcore.ObjectEncoder) error {
 	return enc.AddObject("handler", p.HandlerSpec)
 }
 
-// Less orders procedures lexicographically on (Service, Name).
+// Less orders procedures lexicographically on (Service, Name, Encoding).
 func (p Procedure) Less(other Procedure) bool {
+	if p.Name == other.Name {
```

Comment:
this condition ignores the service... so if I have two services with the same schema it will fail to sort properly.